pkg/client/outofband/client.go

```diff
@@ -92,14 +93,32 @@ func (c *Client) CreateRequest(opts ...RequestOptions) (*Request, error) {
 	req.ID = uuid.New().String()
 	req.Type = RequestMsgType
 
-	err := c.connRecorder.SaveInvitation(req.ID, req)
+	err := c.oobService.SaveRequest(req.Request)
 	if err != nil {
-		return nil, fmt.Errorf("failed to save request : %w", err)
+		return nil, fmt.Errorf("outofband service failed to save request : %w", err)
 	}
 
 	return req, nil
 }
 
+// AcceptRequest from another agent and return the ID of a new connection record.
+func (c *Client) AcceptRequest(r *Request) (string, error) {
+	connID, err := c.oobService.AcceptRequest(&outofband.Request{
+		ID:       r.ID,
+		Type:     r.Type,
```

Comment:
It makes sense to check that `r *Request` is not nil.

Reply:
I generally stay away from that extreme level of defensive programming. I don't believe it's one of this project's guidelines.

Reply:
I also generally do not think we need to cover developer errors like passing a nil request. (we should be covering runtime issues.)

Reply:
got it